requirements_test_min.txt

```diff
@@ -9,4 +9,4 @@ pytest-timeout~=2.3
 towncrier~=23.11
 requests
 # Voluntary for test purpose, not actually used in prod, see #8904
-setuptools==41.6.0
+setuptools;python_version>='3.12'
```

Comment:
I'm in [favor](https://github.com/pylint-dev/pylint/pull/8904#discussion_r1284632741) of removing the version pin, but can I ask why limit the python version?

Reply:
Until `3.12`, `setuptools` was bundled with Python itself.

Reply:
This is the test dependencie set to this specific version as a regression test (see comment above) I think the threat reported is not applicable in this case.

Reply:
Got it, I must manually uninstalled it to reproduce that issue 😅